coding_standards.md

```diff
@@ -0,0 +1,95 @@
+# Coding standards
+
+These are common standards that apply to coding.
+
+Mostly these follow common [programming principles](https://en.wikipedia.org/wiki/Category:Programming_principles)
+
+## Rationale
+
+- Ensure consistency across coding standards for different languages
+
+## Standards
+
+### All code is open
+
+Whilst we acknowledge that there are cases where it may be necessary for code to not be open, this standard ensure that those cases are amanged as exceptions.
+
+### All coding is done on a branch
+
+No code changes are made directly to the master branch.
+This reflects the 'golden' state of the branch and mirrors the general standard that changes are never made directly to production services.
+
+### All code is checked for correct coding style
+
+Language-dependent
+
+### All branches have CI
```

Comment:
Wouldn't say all branches have CI.  If you're doing a shared feature branch or bug fix branch, you wouldn't necessarily want it to be linked to a CI build or deployment.  You'd want the CI linked to master and perhaps main development or release branches

Reply:
I think it depends what you mean by CI. I would expect *all* branches to at least run a linting, build and unit test check as a matter of course.  This wouldn't be a breaking issue, but it does provide instant feedback and also makes the current status of the branch visible to everyone, so we can quickly see that something is still a work in progress.

Reply:
I see.  Previously we've only applied CI and branch policies to master and main development branches to ensure they are "clean" and always in working state.  But I understand your point about public visibility of the status of all branches.

Reply:
If you're on GitHub, and following a branching policy CI just kicks off automatically. You don't get a choice!

I would also argue I want to know if CI is passed on my branch _before_ merging to the main branch to ensure I haven't broken anything.